woocommerce/woocommerce
# Consistent legacy form styling for standard and select2 selects under Checkout/My Account

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:

I've opened this up as a draft to get some feedback, mainly because this is modifying legacy styles that have been around forever.

When working on #52870 I noticed several issues namely:

- Select2 under block themes had weird padding
- Selects had no styling on the account page, but did on product pages
- Selects and Select2 were styled differently
- Form fields were inconsistent in terms of sizes/spacing/borders

To attempt to fix this I've created a new stylesheet where I've pulled in the account page and checkout styles. Theme specific sheets can import this. There are CSS variables to control borders which change between themes.

I've tweaked Select2 styles to remove the themes we don't use, and remove opinionated heights to better fit with theme styles.

Take a look and let me know if you see issues with this @woocommerce/rubik 

Here are some before and afters:

Theme | Before    | After |
| -------- | -------- | ------- |
| 2017  | ![Screenshot 2024-12-06 at 16 32 54](https://github.com/user-attachments/assets/9aef32eb-8aaa-44a6-83ba-8d6cf7114ac1) | ![Screenshot 2024-12-06 at 16 32 24](https://github.com/user-attachments/assets/a8c2c873-153c-412d-a9b2-b3e16c09e77b) |
| 2021 | ![Screenshot 2024-12-06 at 16 33 43](https://github.com/user-attachments/assets/bf78d249-42bb-4fe4-8ef1-a7d2636e6c9b)  | ![Screenshot 2024-12-06 at 16 31 24](https://github.com/user-attachments/assets/1f412ce8-6983-4a2f-a85d-3c2bfc40c9ca) |
| 2025    | ![Screenshot 2024-12-06 at 16 33 24](https://github.com/user-attachments/assets/0c0e09ab-9909-4fb3-b734-80c51786387a) | ![Screenshot 2024-12-06 at 16 32 08](https://github.com/user-attachments/assets/ef3a6161-a21f-4c3b-84ee-d9eac8c94883) |
| Storefront | ![Screenshot 2024-12-06 at 16 33 57](https://github.com/user-attachments/assets/b68b555a-051e-4ac2-a47e-23e06315bf9f) | ![Screenshot 2024-12-06 at 16 31 46](https://github.com/user-attachments/assets/1ab702a9-d4ea-4b34-95dd-5a96a1f0f1ed) |

### How to test the changes in this Pull Request:

Implement this snippet:

```
add_action(
	'woocommerce_init',
	function () {
		woocommerce_register_additional_checkout_field(
			array(
				'id'          => 'mynamespace/number',
				'type'        => 'select',
				'label'       => 'Choose number',
				'location'    => 'address',
				'required'    => false,
				'options'     => array(
					array(
						'value' => 1,
						'label' => 'One',
					),
					array(
						'value' => 7,
						'label' => 'Seven',
					),
				),
				'placeholder' => 'Hello, choose a number!',
			)
		);
	}
);
```

This will add a select box to the edit address screen under my account.

Testing this requires checking how select2 and the above field render across different themes and environments. These are the places you can check the fields:

1. **Shortcode checkout** (most important)
2. **My Account edit address screens**

- Test against block and non-block themes. I tested Storefront, Twenty Ten -> Twenty Twenty Five.
- Test using different browsers. I tested FF, Safari, Arc. There may be some discrepancies due to the nature of select inputs. 



### Changelog entry












<details>

<summary>Changelog Entry Details</summary>

#### Significance




#### Type




#### Message 

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>
